classes/user/Collector.php

```diff
@@ -458,19 +458,29 @@ protected function buildUserGroupFilter(Builder $query): self
             ->when($this->contextIds !== null, fn ($subQuery) => $subQuery->whereIn('ug.context_id', $this->contextIds));
 
         $joinSub = clone $subQuery;
+
         $query
             ->addWhereExistsQuery(
                 $subQuery->when(
                     $this->excludeRoles != null,
-                    // This aggregates a column role_count, which holds an information whether user holds specified role
-                    fn (Builder $subQuery) => $subQuery->leftJoinSub(
-                        $joinSub
-                            ->select('uug.user_id')
-                            ->whereIn('ug.role_id', $this->excludeRoles),
-                        'agr',
-                        'uug.user_id',
-                        'agr.user_id'
-                    )->whereNull('agr.user_id')
+                    fn (Builder $subQuery) =>
+                        DB::connection()->getDoctrineConnection()->getDatabasePlatform() instanceof \Doctrine\DBAL\Platforms\MariaDBPlatform
```

Comment:
If the version for MariaDB works with MySQL, I'd vote to leave just this one and avoid checking the database version.

Reply:
OK, yes, it works, the other one is just a little bit better regarding the performance, I believe...
(Eventually MariaDB will allow the outer references in derived tables one day too...)